src/store/getters/state-getters.ts

```diff
@@ -400,10 +405,20 @@ export const isBusySaving = (state: StateIF): boolean => {
 
 /** Is true when the step is valid. */
 export const isDefineCompanyValid = (state: StateIF): boolean => {
-  // If entity is a Coop, check for Coop Type assignment, flagged valid for non-coop entities.
-  const isValidCoopType = isTypeCoop(state) ? !!getCooperativeType(state) : true
+  if (isTypeCoop(state)) {
+    return (!!getCooperativeType(state) && getDefineCompanyStep(state).valid)
+  }
+  return getDefineCompanyStep(state).valid
+}
 
-  return getDefineCompanyStep(state).valid && isValidCoopType
+/** Is true when the step is valid. */
+export const isRestoreBusinessNameValid = (state: StateIF): boolean => {
+  return (
+    getBusinessNameValid(state) &&
+    getNameTranslationsValid(state) &&
+    getRestorationTypeValid(state) &&
+    getApprovalTypeValid(state)
```

Comment:
I really like this 😄. Instead of calling all the getters in the code when checking if valid, you created a new getter that calls all of them. Learning a lot here!

Reply:
Thanks :)

So there are 3 levels of getters:
- per section (eg, business name)
- per page/step (this one)
- for the overall filing (restoration)

Reply:
@JazzarKarim OK to resolve this conversation?